src/encoded/tests/test_file.py

```diff
@@ -1240,10 +1241,128 @@ def test_track_and_file_facet_info_file_patch_override_fields(
     assert tf_info['experiment_type'] == 'new type'
 
 
+@pytest.fixture
+def dcic_lab(testapp, award):
+    item = {
+        'name': '4dn-dcic-lab',
+        'title': '4DN DCIC, HMS',
+        'status': 'current',
+        'awards': [award['@id']],
+        'uuid': '9780e6b5-507c-4287-b817-103e784587c8'
+    }
+    return testapp.post_json('/lab', item).json['@graph'][0]
+
+
+def test_track_and_file_facet_info_pf_attr_2_dcic_w_expt_lab(
+        testapp, proc_file_json, experiment_data, lab, dcic_lab):
+    """ case where experiment is from a lab and file generated by
+        dcic pipe - dcic is made contributing lab for file - encode is lab
+    """
+    experiment_data['lab'] = proc_file_json['lab'] = lab.get('uuid')
+    proc_file_json['contributing_labs'] = [dcic_lab.get('uuid')]
+    pf = testapp.post_json('/file_processed', proc_file_json).json['@graph'][0]
+    experiment_data['processed_files'] = [pf['@id']]
+    testapp.post_json('/experiment_hi_c', experiment_data, status=201)
+    pf2 = testapp.get(pf['@id']).json
```

Comment:
Not clear to me why these GETs are required here... @@object is returned from the post above so you already have `track_and_facet_info` computed from the original post `pf`? Same question throughout these tests.

I suspect the act of posting the experiment changes the calc prop? In that case what you're doing here works by chance because of the mapping bug we discussed previously in this testing (these tests use default mappings). This makes ES item unique key resolution fail, so instead of getting the cached value of `track_and_facet_info` from ES it is recomputed from the DB and gives (presumably) the correct result below. 

My point is if you ran these tests in a production environment they would probably fail because the GET would be correctly handled by ES and you'd get a cached value for the calc prop. You would need to add a sleep after posting the experiment so reindexing of fp can occur.

Reply:
Interesting - to answer your question - yes posting the experiment with the file linked to it changes the value of the calcprop on the file.  I wonder of the bug you found was actually there by design just for this reason?  These are certainly not the only tests that depend on rapid sequential posts/patches upon which expected changes are then asserted likely sooner than indexing could occur.